manifests/server/installdb.pp

```diff
@@ -27,7 +27,6 @@
       owner   => $mysqluser,
       group   => $::mysql::server::mysql_group,
       mode    => 'u+rw',
-      require => Mysql_datadir[ $datadir ],
```

Comment:
1/ The relationship between the error log file and the datadir used to exist only if `$options['mysqld']['log-error']` is evaluated to true (condition on line 24)

3/ Possible fix: replace `require` with before (since you change the direction of the relationship)

Reply:
This what I went with in the end, a `before` rather than `require` on a new branch (since I made a mess of the old one trying to chase down this bug https://tickets.puppetlabs.com/browse/MODULES-3264).  I don't think that I can reopen this one.